lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/AbstractPostgresKgStore.scala

```diff
@@ -0,0 +1,91 @@
+package io.github.tetherlessworld.mcsapps.lib.kg.stores.postgres
+
+import io.github.tetherlessworld.mcsapps.lib.kg.models.edge.KgEdge
+import io.github.tetherlessworld.mcsapps.lib.kg.models.node.KgNode
+import io.github.tetherlessworld.mcsapps.lib.kg.models.source.KgSource
+import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
+import slick.jdbc.PostgresProfile
+
+
+abstract class AbstractPostgresKgStore(protected val dbConfigProvider: DatabaseConfigProvider) extends HasDatabaseConfigProvider[PostgresProfile] {
+  import profile.api._
+
+  lazy val KgNodes = TableQuery[KgNodeTable]
+  lazy val KgEdges = TableQuery[KgEdgeTable]
+  lazy val KgSources = TableQuery[KgSourceTable]
+
+  private class KgNodeTable(tag: Tag) extends Table[KgNode](tag, "KgNodes") {
+    def id = column[String]("ID", O.PrimaryKey)
+
+    //    def inDegree = column[Option[Int]]("IN_DEGREE")
+    //    def labels = column[String]("LABELS")
+    //    def outDegree = column[Option[Int]]("OUT_DEGREE")
+    //    def pageRank = column[Option[Double]]("PAGERANK")
+    def pos = column[Option[Char]]("POS")
+
+    //    def sourceIds = column[String]("SOURCE_IDS")
+    def wordNetSenseNumber = column[Option[Int]]("WORD_NET_SENSE_NUMBER")
+
+    type KgNodeTableColumnTypes = (String, Option[Char], Option[Int])
+
+    def * = (id, pos, wordNetSenseNumber) <> (create, extract.lift)
+
+    private def create: KgNodeTableColumnTypes => KgNode = {
+      case (id, pos, wordNetSenseNumber) =>
+        KgNode(id, None, List(), None, None, pos, List(), wordNetSenseNumber)
+    }
+
+    private def extract: PartialFunction[KgNode, KgNodeTableColumnTypes] = {
+      case KgNode(id, inDegree, labels, outDegree, pageRank, pos, sourceIds, wordNetSenseNumber) =>
+        (id, pos, wordNetSenseNumber)
+    }
+  }
+
+  class KgEdgeTable(tag: Tag) extends Table[KgEdge](tag, "KgEdges") {
+    def id = column[String]("ID", O.PrimaryKey)
+//    labels: List[String],
+    def objectNodeId = column[String]("OBJECT_NODE_ID")
```

Comment:
The s, p, o should form a composite unique constraint. Fine to do that later.

Reply:
Cool, I will keep this in mind

Reply:
Added a `unique_constraint` to `kg_edge` https://github.com/tetherless-world/mcs-apps/blob/0d12ee70978280c2a69a72b7a072d23fe8f79461/lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/AbstractPostgresKgStore.scala#L30